pandas/tests/test_window.py

```diff
@@ -3801,3 +3801,20 @@ def test_non_monotonic(self):
         df2 = df.sort_values('B')
         result = df2.groupby('A').rolling('4s', on='B').C.mean()
         tm.assert_series_equal(result, expected)
+
+    def test_rolling_cov_offset(self):
+        # GH16058
+
+        idx = pd.date_range('2017-01-01', periods=24, freq='1h')
+        ss = pd.Series(np.arange(len(idx)), index=idx)
+
+        result = ss.rolling('2h').cov()
```

Comment:
also test again using ``ss.rolling(2).cov()`` (will be the same for this case).

I don't know if we have any tests for a non-regular date range (Iow where freq='2' and freq='2h' are actually different.